From 5679a0bf6ef6db2568afd7d23f9e0b372c0b5a00 Mon Sep 17 00:00:00 2001
From: Astro <astro@spaceboyz.net>
Date: Fri, 15 Aug 2008 16:47:30 +0200
Subject: [PATCH] Allow to initialize NS with prefix=nil (aka default namespace)

---
 ext/libxml/ruby_xml_ns.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/ext/libxml/ruby_xml_ns.c b/ext/libxml/ruby_xml_ns.c
index 8c3c16c..517fccf 100644
--- a/ext/libxml/ruby_xml_ns.c
+++ b/ext/libxml/ruby_xml_ns.c
@@ -26,9 +26,11 @@ VALUE
 ruby_xml_ns_initialize(VALUE self, VALUE node, VALUE href, VALUE prefix) {
   xmlNodePtr xnode;
   xmlNsPtr xns;
+  xmlChar *prefixS;
 
   Data_Get_Struct(node, xmlNode, xnode);
-  xns = xmlNewNs(xnode, (xmlChar*)StringValuePtr(href), (xmlChar*)StringValuePtr(prefix));
+  prefixS = NIL_P(prefix) ? NULL : StringValuePtr(prefix);
+  xns = xmlNewNs(xnode, (xmlChar*)StringValuePtr(href), prefixS);
 
   DATA_PTR(self) = xns;
   return self;  
-- 
1.5.6.3

